### PR TITLE
Build pipeline fix to allow for older versions of MacOS

### PR DESCRIPTION
**Pull request by gcampbell-msft**
Edits the build.yml to use specific versions of python targeting mac such that older versions of mac can be supported.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -32,14 +36,20 @@ jobs:
         architecture: 'x86'
     - name: Set up Python 3.8 x64
       uses: actions/setup-python@v2
-      if: matrix.os == 'ubuntu-latest' || matrix.os == 'macos-latest'
+      if: ${{ matrix.os == 'ubuntu-latest' }}
```
Comment (benmcmorran) [inline .github/workflows/build.yml]: Braces shouldn't be needed here. The conditional is already an expression.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -32,14 +36,20 @@ jobs:
...
       with:
         python-version: 3.8
         architecture: 'x64'
+    - name: Set up Python for macOS
+      if: ${{ matrix.os == 'macos-latest' }}
```
Comment (benmcmorran) [inline .github/workflows/build.yml]: Same here, braces aren't required.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -32,14 +36,20 @@ jobs:
...
+      run: |
```
Comment (benmcmorran) [inline .github/workflows/build.yml:47]: I'd add a comment explaining why we aren't using `actions/setup-python` here.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -32,14 +36,20 @@ jobs:
...
+        curl https://www.python.org/ftp/python/${{ matrix.PYTHON_VERSION }}/python-${{ matrix.PYTHON_VERSION }}-macosx10.9.pkg -o "python.pkg"
```
Comment (benmcmorran) [inline .github/workflows/build.yml:48]: Can we validate the hash of this file after the download? I think `shasum` will already be installed on the Mac agents.

**Review by benmcmorran — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (gcampbell-msft) [reply .github/workflows/build.yml]: fixed.

File: .github/workflows/build.yml (hunk 2 above)
Comment (gcampbell-msft) [reply .github/workflows/build.yml]: fixed.

File: .github/workflows/build.yml (hunk 3 above)
Comment (gcampbell-msft) [reply .github/workflows/build.yml:47]: fixed.

File: .github/workflows/build.yml (hunk 4 above)
Comment (gcampbell-msft) [reply .github/workflows/build.yml:48]: Yes, inserted the code to do this.

**Pull request merged**